libparsec/crates/client/src/client/tos.rs

```diff
@@ -0,0 +1,98 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::collections::HashMap;
+
+use libparsec_client_connection::ConnectionError;
+use libparsec_protocol::tos_cmds;
+use libparsec_types::prelude::*;
+
+use super::Client;
+
+#[derive(Debug, thiserror::Error)]
+pub enum ClientGetTosError {
+    #[error("There is no Terms of Service for the organization")]
+    NoTos,
+    #[error("Cannot reach the server")]
+    Offline,
+    #[error(transparent)]
+    Internal(#[from] anyhow::Error),
+}
+
+impl From<ConnectionError> for ClientGetTosError {
+    fn from(value: ConnectionError) -> Self {
+        match value {
+            ConnectionError::NoResponse(_) => Self::Offline,
+            // TODO: handle organization expired and user revoked here ?
+            err => Self::Internal(err.into()),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+/// Terms of Service
+pub struct Tos {
+    pub per_locale_url: HashMap<String, String>,
```

Comment:
We could consider the default ToS being the first entry in the map

Reply:
The locale format (i.e. `xx_YY`) is standard so I would say it is obvious enough with the current documentation of `--organization-initial-tos` when doing `parsec run --help`

For URL, I think we don't want to limit to any format given we should only display the URL in the Parsec GUI and allow the user to click on it to open it in his own browser.

Finaly regarding the default locale, I think it would be a good idea to idea to document something like
> if the locale of a client is not among the ones defined, the first specified locale will be used as fallback

However we must first ensure this is the actual behavior in the GUI (@Max-7 what do you think about that ?)
And we should also make sure the dict order is not modified (the good news is it would mean all my suffering to implement the javascript bindings with Map instead of object [were not in vain](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#objects_vs._maps)  ^^)

Reply:
`BTreeMap` is not the solution to keep the order of insertion, it will just order the keys by what `Ord` indicate.

> If we want to go to the order route, use a `Vec<Lang, URL>`

IMO, the problem of which TOS to use is one the client side, so is it really important to provide guarantee about the order of TOS' URL ?

If that default TOS is important, best would be to add an additional fields like `default_tos` and `default_tos_lang`.

Reply:
I agree that _order_ is not needed here. 

The key question is: do we need a default value? 
- If the GUI presents a single TOS to the user, then it only needs a fall-back mechanism (like the `default_tos_lang`).
- If the GUI presents all available TOS (e.g. a combo allows to change lang), then there is probably no need to define a default value.

Reply:
Just provide everything you have, the GUI will sort it out. It knows what locale the current user prefers, it can decide what to display, there's no need for a default value. Worst case, if it is absolutely not able to guess what locale the user would prefer among the ones provided, it will just use the first one from the list.

Reply:
So no order and no default, I like that !